components/eventing-controller/controllers/subscription/nats/reconciler_test.go

```diff
@@ -116,161 +116,80 @@ func testCreateDeleteSubscription(id int, eventTypePrefix, natsSubjectToPublish,
-			changedSub = givenSubscription.DeepCopy()
-			changedSub.Spec.Sink = fmt.Sprintf("https://%s.%s.svc.cluster.local", "testapp", "test-ns")
-			Expect(k8sClient.Update(ctx, changedSub)).Should(BeNil())
-
-			getSubscription(ctx, givenSubscription).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveCondition(eventingv1alpha1.MakeCondition(
-					eventingv1alpha1.ConditionSubscriptionActive,
-					eventingv1alpha1.ConditionReasonNATSSubscriptionActive,
-					v1.ConditionFalse, "namespace of subscription: test and the namespace of subscriber: test-ns are different")),
-			))
-
-			subscriptionEvent = v1.Event{
-				Reason:  string(events.ReasonValidationFailed),
-				Message: "Namespace of subscription: test and the subscriber: test-ns are different",
-				Type:    v1.EventTypeWarning,
-			}
-			getK8sEvents(&subscriptionEvents, givenSubscription.Namespace).Should(reconcilertesting.HaveEvent(subscriptionEvent))
-
-			By("Updating the subscription configuration in the spec with invalid sink URL with non-existing service name")
-			changedSub = givenSubscription.DeepCopy()
-			reconcilertesting.WithValidSink(namespaceName, "testapp", changedSub)
-			Expect(k8sClient.Update(ctx, changedSub)).Should(BeNil())
-
-			getSubscription(ctx, givenSubscription).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveCondition(eventingv1alpha1.MakeCondition(
-					eventingv1alpha1.ConditionSubscriptionActive,
-					eventingv1alpha1.ConditionReasonNATSSubscriptionActive,
-					v1.ConditionFalse, "sink is not valid cluster local svc, failed with error: Service \"testapp\" not found")),
-			))
-
-			subscriptionEvent = v1.Event{
-				Reason:  string(events.ReasonValidationFailed),
-				Message: "Sink does not correspond to a valid cluster local svc",
-				Type:    v1.EventTypeWarning,
-			}
-			getK8sEvents(&subscriptionEvents, givenSubscription.Namespace).Should(reconcilertesting.HaveEvent(subscriptionEvent))
-
-			By("Updating the subscription configuration in the spec with invalid sink URL with valid subscriber service name")
-			changedSub = givenSubscription.DeepCopy()
-
-			subscriberName := fmt.Sprintf(subscriberNameFormat, id)
-			subscriberSvc := reconcilertesting.NewSubscriberSvc(subscriberName, namespaceName)
-			ensureSubscriberSvcCreated(ctx, subscriberSvc)
-
-			reconcilertesting.WithValidSink(namespaceName, subscriberSvc.Name, changedSub)
-			Expect(k8sClient.Update(ctx, changedSub)).Should(BeNil())
-
-			getSubscription(ctx, givenSubscription).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveCondition(eventingv1alpha1.MakeCondition(
-					eventingv1alpha1.ConditionSubscriptionActive,
-					eventingv1alpha1.ConditionReasonNATSSubscriptionActive,
-					v1.ConditionTrue, "")),
-			))
+func testCreateSubscriptionWithInvalidSink(id int, eventTypePrefix, _, eventTypeToSubscribe string) bool {
```

Comment:
Perhaps we should have a "positive" test too, something like:
```
 getSubscriptionWithValidSink(...)
 ```
 which should avoid issues like this: https://github.com/kyma-project/kyma/issues/12810
 If you think that's another issue, please create a new one and link it

Reply:
@radufa positive test is already covered in this [test](https://github.com/kyma-project/kyma/blob/d91cde56f0bda1504e4a2e9e8869f822c5e49823/components/eventing-controller/controllers/subscription/nats/reconciler_test.go#L93)